app/upgrades/signetlaunch/upgrades.go

```diff
@@ -79,6 +101,115 @@ func LoadBTCHeadersFromData() ([]*btclighttypes.BTCHeaderInfo, error) {
 	return gs.BtcHeaders, nil
 }
 
+// LoadSignedFPsFromData returns the finality providers from the json string.
+// It also verifies if the msg is correctly signed and is valid to be inserted.
+func LoadSignedFPsFromData(cdc codec.Codec, txJSONDecoder sdk.TxDecoder) ([]*btcstktypes.MsgCreateFinalityProvider, error) {
+	buff := bytes.NewBufferString(SignedFPsStr)
+
+	var d dataSignedFps
+	err := json.Unmarshal(buff.Bytes(), &d)
+	if err != nil {
+		return nil, err
+	}
+
+	fps := make([]*btcstktypes.MsgCreateFinalityProvider, len(d.SignedTxsFP))
+	for i, txAny := range d.SignedTxsFP {
+		txBytes, err := json.Marshal(txAny)
+		if err != nil {
+			return nil, err
+		}
+
+		tx, err := txJSONDecoder(txBytes)
+		if err != nil {
+			return nil, err
+		}
+
+		fp, err := parseCreateFPFromSignedTx(cdc, tx)
+		if err != nil {
+			return nil, err
+		}
+
+		fps[i] = fp
+	}
+
+	// sorts all the FPs by their addresses
+	sort.Slice(fps, func(i, j int) bool {
+		return fps[i].Addr > fps[j].Addr
+	})
+
+	return fps, nil
+}
+
+func parseCreateFPFromSignedTx(cdc codec.Codec, tx sdk.Tx) (*btcstktypes.MsgCreateFinalityProvider, error) {
```

Comment:
should we also check the commission rate to be not less than the min value defined in params?

Reply:
The commission rate is already check at https://github.com/babylonlabs-io/babylon/blob/1b6e6bafea91db6262dc7b40ae0875ad069023ad/x/btcstaking/keeper/finality_providers.go#L23